### PR TITLE
switch from vector to hashmap for headers in handleRequest

### PR DESCRIPTION
**Pull request by karthik2804**
Signed-off-by: karthik Ganeshram <karthik.ganeshram@fermyon.com>

### CODE REVIEW

File: crates/spin-js-engine/src/lib.rs [hunk 1]
```diff
@@ -447,20 +457,18 @@ fn handle(request: Request) -> Result<Response> {
     process.set_property("env", env)?;
 
     global.set_property("process", process)?;
+    
+    let headers = context.object_value()?;
+    for (key, value) in request.headers().iter() {
+        let header_value =  context.value_from_str(&str::from_utf8(value.as_bytes()).unwrap().to_owned());
```
Comment (dicej) [inline crates/spin-js-engine/src/lib.rs]: I think we should probably just let the error bubble up here (which will avoid the need for the `if let` below).
```suggestion
        let header_value =  context.value_from_str(&str::from_utf8(value.as_bytes()).unwrap().to_owned())?;
```

Comment (karthik2804) [reply crates/spin-js-engine/src/lib.rs]: fixed it. Removed the unwrap as well.

File: crates/spin-js-engine/src/lib.rs [hunk 2]
```diff
@@ -447,20 +457,16 @@ fn handle(request: Request) -> Result<Response> {
     process.set_property("env", env)?;
 
     global.set_property("process", process)?;
+    
+    let headers = context.object_value()?;
+    for (key, value) in request.headers().iter() {
+        let header_value =  context.value_from_str(&str::from_utf8(value.as_bytes()).to_owned()?)?;
```
Comment (dicej) [inline crates/spin-js-engine/src/lib.rs]:
```suggestion
        let header_value =  context.value_from_str(str::from_utf8(value.as_bytes())?)?;
```

Comment (karthik2804) [reply crates/spin-js-engine/src/lib.rs]: fixed

**Review by dicej — APPROVED** [review]

**Pull request merged**